### PR TITLE
Fix/sensitive output

### LINKED ISSUE #6: [BUG] Sensitive output
### Community Note

The sensitive parameter is required for the output.

Error:
```
Error: Output refers to sensitive values

  on outputs.tf line 1:
   1: output "linux_function_app" {
```

Solution: Add ```sensitive = true``` to the output
```hcl
output "linux_function_app" {
  description = "Linux Function App output object if Linux is chosen. Please refer to `./modules/linux-function/README.md`"
  value       = try(module.linux_function["enabled"], null)
  sensitive   = true
}

output "windows_function_app" {
  description = "Windows Function App output object if Windows is chosen. Please refer to `./modules/windows-function/README.md`"
  value       = try(module.windows_function["enabled"], null)
  sensitive   = true
}
```

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.3.9

### AzureRM Provider Version

3.92.0

### Affected Resource(s)/Data Source(s)

azurerm_linux_function_app.linux_function

### Terraform Configuration Files

```hcl
module "azure_region" {
  source  = "claranet/regions/azurerm"
  version = "x.x.x"

  azure_region = var.azure_region
}

module "rg" {
  source  = "claranet/rg/azurerm"
  version = "x.x.x"

  location    = module.azure_region.location
  client_name = var.client_name
  environment = var.environment
  stack       = var.stack
}

module "logs" {
  source  = "claranet/run/azurerm//modules/logs"
  version = "x.x.x"

  client_name         = var.client_name
  environment         = var.environment
  stack               = var.stack
  location            = module.azure_region.location
  location_short      = module.azure_region.location_short
  resource_group_name = module.rg.resource_group_name
}

### Linux
module "function_app_linux" {
  source  = "claranet/function-app/azurerm"
  version = "x.x.x"

  client_name         = var.client_name
  environment         = var.environment
  stack               = var.stack
  location            = module.azure_region.location
  location_short      = module.azure_region.location_short
  resource_group_name = module.rg.resource_group_name

  name_prefix = "hello"

  os_type              = "Linux"
  function_app_version = 4
  function_app_site_config = {
    application_stack = {
      python_version = "3.9"
    }
  }

  function_app_application_settings = {
    "tracker_id"      = "AJKGDFJKHFDS"
    "backend_api_url" = "https://backend.domain.tld/api"
  }

  storage_account_identity_type = "SystemAssigned"

  logs_destinations_ids = [
    module.logs.logs_storage_account_id,
    module.logs.log_analytics_workspace_id
  ]

  extra_tags = {
    foo = "bar"
  }
}
```

### Debug Output/Panic Output

```shell
Error: Output refers to sensitive values

  on outputs.tf line 1:
   1: output "linux_function_app" {

To reduce the risk of accidentally exporting sensitive data that was intended
to be only internal, Terraform requires that any root module output
containing sensitive data be explicitly marked as sensitive, to confirm your
intent.

If you do intend to export this data, annotate the output value as sensitive
by adding the following argument:
    sensitive = true
```

### Expected Behaviour

_No response_

### Actual Behaviour

_No response_

### Steps to Reproduce

_No response_

### Important Factoids

_No response_

### References

_No response_

### PR DESCRIPTION
**Pull request by joaolms**
Fixes #6.

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Changes proposed in this pull request

- Add ```sensitive = true``` for linux and windows functions

@claranet/fr-azure-reviewers

### CODE REVIEW

**Review by jmapro — APPROVED** [review]
LGTM

File: outputs.tf [hunk 1]
```diff
@@ -1,11 +1,13 @@
 output "linux_function_app" {
   description = "Linux Function App output object if Linux is chosen. Please refer to `./modules/linux-function/README.md`"
   value       = try(module.linux_function["enabled"], null)
+  sensitive   = true
```
Comment (Shr3ps) [inline outputs.tf:4]: Hum no, I fully disagree with this.
It will prevent using correctly the module and access properties in a stack context with Terraform or OpenTofu.

**Review by Shr3ps — CHANGES_REQUESTED** [review]

**Comment by Shr3ps** [comment]
> LGTM

nope, I think we don't want this.
@joaolms please give more details in your opened issue, and how you got the error exactly.

**Comment by kayahk** [comment]
We had the same issue with the missing `sensitive = true` argument in the `eventhub` module and now we see it here in the `function-app` module, too.
```
module.linux_function["enabled"].module.diagnostics.azurerm_monitor_diagnostic_setting.main[0]: Creation complete after 4s [id=/subscriptions/f7c1a591-6b7c-4aee-b0aa-3ef509b9218a/resourceGroups/rg-sgc-sus-ops1-germanywestcentral/providers/Microsoft.Web/sites/fun-sgc-sus-ops1-germanywestcentral|default]
╷
│ Error: Output refers to sensitive values
│ 
│   on outputs.tf line 1:
│    1: output "linux_function_app" {
│ 
│ To reduce the risk of accidentally exporting sensitive data that was
│ intended to be only internal, Terraform requires that any root module
│ output containing sensitive data be explicitly marked as sensitive, to
│ confirm your intent.
│ 
│ If you do intend to export this data, annotate the output value as
│ sensitive by adding the following argument:
│     sensitive = true
```
I can't exactly follow why adding the `sensitive` argument would prevent using the module correctly - in fact, it would make the module usable (at all) for us in the first place. It shouldn't be a problem to keep existing references to outputs, they wouldn't be impacted by changing from non-sensitive to sensitive.

**Comment by Shr3ps** [comment]
> We had the same issue with the missing `sensitive = true` argument in the `eventhub` module and now we see it here in the `function-app` module, too.
> 
> ```
> module.linux_function["enabled"].module.diagnostics.azurerm_monitor_diagnostic_setting.main[0]: Creation complete after 4s [id=/subscriptions/f7c1a591-6b7c-4aee-b0aa-3ef509b9218a/resourceGroups/rg-sgc-sus-ops1-germanywestcentral/providers/Microsoft.Web/sites/fun-sgc-sus-ops1-germanywestcentral|default]
> ╷
> │ Error: Output refers to sensitive values
> │ 
> │   on outputs.tf line 1:
> │    1: output "linux_function_app" {
> │ 
> │ To reduce the risk of accidentally exporting sensitive data that was
> │ intended to be only internal, Terraform requires that any root module
> │ output containing sensitive data be explicitly marked as sensitive, to
> │ confirm your intent.
> │ 
> │ If you do intend to export this data, annotate the output value as
> │ sensitive by adding the following argument:
> │     sensitive = true
> ```
> 
> I can't exactly follow why adding the `sensitive` argument would prevent using the module correctly - in fact, it would make the module usable (at all) for us in the first place. It shouldn't be a problem to keep existing references to outputs, they wouldn't be impacted by changing from non-sensitive to sensitive.

Adding `sensitive` to a complete resource outputted (which is not a "simple" value), of type of map/object is against Terraform best practices, it will prevent to be able to easily use outputs attributes from one module to another directly, it will requires the usage of `nonsensitive` on many output calls.

If you read again your error message, it seems that with Terragrunt and in your context this repository **is not used as a module** like it should be in the Terraform/OpenTofu ecosystem. It seems the repo is used as a "root module" (also called a Terraform Stack: https://www.hashicorp.com/blog/terraform-stacks-explained) but it should not.

So to be able to correctly use this module, like many others (for example official Azure one: https://github.com/Azure/terraform-azurerm-avm-res-eventhub-namespace/blob/main/outputs.tf) you should consider using the standard Terraform/OpenTofu way to call modules.

**Comment by kayahk** [comment]
@Shr3ps thanks for your comment, I understand your position regarding the opinionated approach following Terraform (or OpenTofu). Unfortunately, a stack in Terragrunt can indeed be a collection of modules - regardless if root or not - since Terragrunt itself acts as the wrapping orchestrator instead of the `stack` concept propagated by Hashicorp. We don't have any issues with referencing output attributes across the entire stack, while it is true that the references can be somewhat easier to lookup with non-sensitive primitive types.
Bottom line is that we have to look for another way to implement the modules in question, that's all.
